public/app/plugins/datasource/loki/datasource.ts

```diff
@@ -348,7 +350,24 @@ export class LokiDatasource
     return Array.from(streams);
   }
 
-  // By implementing getTagKeys and getTagValues we add ad-hoc filtters functionality
+  async getDataSamples(query: LokiQuery): Promise<DataFrame[]> {
+    // Currently works only for log samples
+    if (!isValidQuery(query.expr) || !isLogsQuery(query.expr)) {
+      return [];
+    }
+
+    const lokiLogsQuery: LokiQuery = {
+      expr: query.expr,
+      queryType: LokiQueryType.Range,
+      refId: 'log-samples',
+      maxLines: 10,
+    };
+
+    const request = makeRequest(lokiLogsQuery, this.timeSrv.timeRange(), CoreApp.Explore, 'log-samples');
```

Comment:
We considered here to not run this as a:
-  query
- resource call. 

### query
Disadvantage of `query` is that it is going to appear in Inspector. Moreover, getting samples is basically resource call - we just need samples as resources. However, this is currently the way how to get response in dataframes.

### resource call
If we would run this as resource call, we wouldn't get the dataframe, but Loki response which we would have to turn into dataframe. This would mean that we would have to hold translation logic at frontend and backend, which we don't want. We can re-evaluate this if Loki team decided to change Loki's response to dataframe.